NAACCR/manual work/manual_work_08-Feb-23.sql

```diff
@@ -0,0 +1,171 @@
+FROM concept_manual_backup_2023_02_08
+;
+
+--CHECK THE # OF CODES OVERLAPPING BETWEEN MANUAL REFRESH AND PREVIOUS CRM version in case of identical relationships and thier validities
+SELECT *
+FROM concept_relationship_manual_backup_2023_02_08 crmb
+WHERE EXISTS (
+    SELECT 1
+    from concept_relationship_manual_refresh crmr
+    WHERE crmr.concept_code_1=crmb.concept_code_1
+    and crmr.vocabulary_id_1=crmb.vocabulary_id_1
+    and crmr.relationship_id=crmb.relationship_id
+          )
+and crmb.invalid_reason IS NULL
+and crmb.concept_code_1<>crmb.concept_code_2
+and crmb.vocabulary_id_1<>crmb.vocabulary_id_2
+;
+
+--1toM check
+-- 334 rows including simultaneous maps to Descendant and Ancestor- TBD
+--Clinically relevant
+SELECT DISTINCT *
+   FROM concept_relationship_manual_refresh
+JOIN concept c ON concept_relationship_manual_refresh.concept_code_2 = c.concept_code
+and c.vocabulary_id='Cancer Modifier'
+    WHERE (concept_code_1,relationship_id) IN (
+        SELECT concept_code_1,relationship_id
+   FROM concept_relationship_manual_refresh
+   GROUP BY concept_code_1,relationship_id HAVING count(DISTINCT concept_code_2)>1
+        )
+;
+
+
+
+
+--CRM process
+TRUNCATE concept_relationship_manual;
+INSERT INTO concept_relationship_manual (concept_code_1, vocabulary_id_1, relationship_id, valid_start_date,
+                                         valid_end_date, invalid_reason, concept_code_2, vocabulary_id_2)
+SELECT DISTINCT
+       concept_code_1,
+       vocabulary_id_1,
+       relationship_id,
+       valid_start_date,
+       valid_end_date,
+       invalid_reason,
+       concept_code_2,
+       vocabulary_id_2
+FROM concept_relationship_manual_backup_2023_02_08
+;
+
+
+--Detect codes not existing as naaccr values
+SELECT *
+FROM concept_relationship_manual_refresh
+WHERE concept_code_1 not in (SELECT concept_code from concept where concept_class_id='NAACCR Value')
+;
+
+--CHeck that NAACCR Values are not target for other codes
+SELECT *
+FROM concept_relationship_manual_refresh a
+JOIN concept b
+on a.concept_code_1=b.concept_code and concept_class_id='NAACCR Value'
+JOIN concept_relationship r
+on b.concept_id=r.concept_id_2
+and r.relationship_id='Maps to'
+and r.invalid_reason is null
+and r.concept_id_2<>r.concept_id_1;
+
+
+
+
+--Mapping insertion
+INSERT INTO concept_relationship_manual (concept_code_1, vocabulary_id_1,  relationship_id, valid_start_date, valid_end_date, invalid_reason,concept_code_2,vocabulary_id_2)
+SELECT concept_code_1, vocabulary_id_1,  relationship_id, CURRENT_DATE as valid_start_date, valid_end_date, invalid_reason,concept_code_2,vocabulary_id_2
+FROM concept_relationship_manual_refresh
+where (concept_code_1,relationship_id,concept_code_2) NOT IN (SELECT concept_code_1,relationship_id,concept_code_2 from concept_relationship_manual where invalid_reason IS NULL);
+;
+
+--Set Non-standard concept class
+UPDATE concept_manual cm
```

Comment:
@vladkorsik We need to move this logic to the load_stage going forward

Reply:
@Alexdavv Personally I do not want to mov it their as it relyses on _refresh tables

Reply:
@vladkorsik Pleas add this to the abovementioned issue

Reply:
i would like to have enough time to refactor the LS since in reality is was the greatest EXCEL based community contribution ever. Not possible currently